homeassistant/components/aosmith/config_flow.py

```diff
@@ -32,30 +56,56 @@ async def async_step_user(
             await self.async_set_unique_id(unique_id)
             self._abort_if_unique_id_configured()
 
-            session = aiohttp_client.async_get_clientsession(self.hass)
-            client = AOSmithAPIClient(
-                user_input[CONF_EMAIL], user_input[CONF_PASSWORD], session
+            error = await self._async_validate_credentials(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
             )
-
-            try:
-                await client.get_devices()
-            except AOSmithInvalidCredentialsException:
-                errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
-            else:
+            if error is None:
                 return self.async_create_entry(
                     title=user_input[CONF_EMAIL], data=user_input
                 )
 
+            errors["base"] = error
+
         return self.async_show_form(
             step_id="user",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_EMAIL): str,
+                    vol.Required(CONF_EMAIL, default=self._reauth_email): str,
```

Comment:
`self._reauth_email` doesn't appear to be set to something else than `None` when in the user flow.